kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/utils/SerializationTest.java

```diff
@@ -242,10 +244,10 @@ void unmarshalWithInvalidYamlShouldThrowException() {
   }
 
   @Test
-  @DisplayName("unmarshal, with invalid YAML resource, should throw exception")
-  void unmarshalWithInvalidResourceShouldThrowException() {
```

Comment:
This reverts #4263 / #4206

Are we sure we want to proceed with this change? In which cases are we going to use our Serialization tooling to unmarshal an "invalid"/incomplete resource?

Reply:
> This reverts https://github.com/fabric8io/kubernetes-client/pull/4263 / https://github.com/fabric8io/kubernetes-client/issues/4206

The main qualm addressed with throwing an exception with 4263 was not returning an untyped GenericKubernetesResource, which is also HasMetadata.  Here we're returning a RawExtension - so there's no ambiguity.

Reply:
:thinking: 

I feel that the following methods are still ambiguous:
```java
public static <T> T unmarshal(InputStream is) {
// ...
public static <T> T unmarshal(String str) {
```

We probably need to clean up this class (other PR/Issue).

I see value in a `Serialization.unmarshall` field that can return any instance of HasMetadata, list of HasMetadata, or maybe even KubernetesList, but only restricted for these types.

The current behavior for the referenced methods is sort of unpredictable.

Reply:
> I feel that the following methods are still ambiguous:

That's a different level of ambiguity.  What I'm mostly trying to do is remove special case handling.  If we have no type specified, then RawExtension makes just as much sense as throwing an exception - but allows us to remove the specialized check in Serialization.

> I see value in a Serialization.unmarshall field that can return any instance of HasMetadata, list of HasMetadata, or maybe even KubernetesList, but only restricted for these types.

That seems fine to add - Serialization.unmarshallHasMetadata or another signature with a hasmetadata flag.

Related to this - a yaml list of HasMetadata is currently problematic.  The root unmarshall method conversion from yaml only accounts for an object root - an array will fail with a class cast.